### ISSUE TITLE
Publish the next Java debian package in EV3Dev

### ISSUE THREAD

**Issue by jabrena** [ev3dev/ev3dev#1437]
Hi @dlech,

In relation to the [Java packaging for EV3Dev](https://github.com/ev3dev/ev3dev/issues/1099), I would like to improve the traceability of this process. Currently, the process is manual and it doesn´t have any traceability in terms of issues and documentation. In the past, @JakubVanek was collaborating in the project but currently, he is not an active committer for [ev3dev-lang-java](https://github.com/ev3dev-lang-java) so I will replace him for this particular process.

**For Java 11, current Debian package published will be the last one and I will prepare the next one for Java 17 LTS.**

**Tasks:**

- [ ] Clarify the process to upgrade the Java Debian package for Stretch & Buster distro for EV3Dev
- [x] Determinate what stable Java Debian package is available for Stretch & Buster distro for EV3Dev
- [ ] Review the documentation to make a Build for Java 15
- [ ] Review the documentation to make a Patch for current Builds

Many thanks in advance

Juan Antonio

**Comment by JakubVanek** [comment]
Hi all,

I agree with the revocation of my maintainer status on the [`jri-11-ev3`](https://github.com/ev3dev-lang-java/openjdk-ev3) and [`libopenmpt`](https://github.com/ev3dev/libopenmpt) packages. As I do not maintain other packages, I agree with the revocation of my SFTP credentials as well.

To prevent confusion in the future, I'd like to clarify here what package versions are published now.

* `libopenmpt` has version `0.2.7386~beta20.3-3+deb9u3ev3dev1`, not changed since https://github.com/ev3dev/ev3dev/issues/1204 was fixed.
* `jri-11-ev3` has version `11.0.8~10` uploaded on 2020-08-10, newer than the previous 11.0.6.

I have uploaded the 11.0.8 notwhistanding the previously agreed 11.0.6 indefinite version freeze. The reason for this is me wanting to fix a consequence of bug https://github.com/ev3dev-lang-java/ev3dev-lang-java/issues/731, that is, some previous ev3dev images were generated with only partially configured package `ca-certificates-java` (or similar one). This manifested as the dpkg configuration of `ca-certificates-java` or `jri-11-ev3` packages being finished only after the image was installed on the brick.

To fix the bug, I have picked at the time the latest JRI11 from https://ci.adoptopenjdk.net/view/ev3dev/job/eljbuild and I have packaged it using the fixed https://github.com/ev3dev-lang-java/openjdk-ev3/pull/66 branch (with using Mozilla's `cacerts`). This should make the package configuration succeed when it is [first installed](https://github.com/ev3dev/docker-library/blob/01c68dff78043d58a30fb31d669db8dd4e3e5964/ev3dev-stretch/layers/generic/brickstrap/generic/run#L63) and it fixes https://github.com/ev3dev-lang-java/ev3dev-lang-java/issues/728 (verified, broker now runs successfully after upgrading jri-11-ev3)..

The reason for picking 11.0.8 and not 11.0.6 is twofold. I have tried to monkey-patch the new CA certificates into the old 11.0.6 package. However, this was a) quite fragile, and b) it did not fix https://github.com/ev3dev-lang-java/ev3dev-lang-java/issues/731 because elliptic-curve crypto module was still missing in that build. Addressing both of these concerns would require having access to the .tar.gz JDK11 archive for 11.0.6, but this was automatically replaced by newer builds on the AdoptOpenJDK Jenkins.

I know I have done this without your knowledge and I accept the consequences of that. My argument for this is that I am not leaving the package in a possibly long lasting state in which it is breaking something external to the project. I did not discuss this because I did not think it would achieve my goal of doing this without giving ev3dev-lang-java more time to implement the AdoptOpenJDK test suite.

Best regards,

Jakub

**Comment by jabrena** [comment]
Hi @JakubVanek,

in this case exist several issues in your behaviour:

- It is not your criteria to take any decision to publish/upgrade the Debian package about Java
- The package uploaded comes from your laptop or another source, because the pipeline doesn´t have the binary that you published
- You avoided the Pull Request process
- Your actions are not traced in any issue backlog: https://github.com/ev3dev-lang-java/ev3dev-lang-java/issues or https://github.com/ev3dev/ev3dev/issues 

So, in summary, you published a code that it doesn´t come from source of truth:
https://ci.adoptopenjdk.net/view/ev3dev/job/eljbuild/job/stretch-11/

Tell me, what is the steps to rollback to previous package version? 
**I would like to do it**

For the future:

- How to check the checksum from the package generated by the pipeline in compare to the package stored into the `ev3dev Debian repository`? I would like to make a test to check that the exact binary is published on the repository and it is maintained.

**Tell me if it is clear the actions to do, you are abusing about the generosity to collaborate in the Java project.**

Juan Antonio

**Comment by JakubVanek** [comment]
> So, in summary, you published a code that it doesn´t come from source of truth:
> https://ci.adoptopenjdk.net/view/ev3dev/job/eljbuild/job/stretch-11/

The base JRI tarball does come from Jenkins, I have not build 11.0.6 locally for this reason. The `metadata` descriptor inside the packages is the following:

* Stretch
```
# ev3dev-lang-java openjdk build metadata
JAVA_ORIGIN="git"
JAVA_REPO="https://github.com/openjdk/jdk11u.git"
JAVA_BRANCH="jdk-11.0.8+10"
JAVA_COMMIT="871fe6e553f9b35f348240eaa5faab39b5de4c1b"
JAVA_VERSION="11.0.8+10-ev3"
CONFIG_VM="client"
CONFIG_DEBUG="release"
CONFIG_VERSION="11"
CONFIG_PLATFORM="ev3"
CONFIG_MODULES="java.se,jdk.jdwp.agent,jdk.unsupported,jdk.management.agent,jdk.jartool,jdk.crypto.ec"
BUILDER_COMMIT="9856847ab129cb9eb448b8e96562583778d76d86"
BUILDER_EXTRA="Jenkins eljbuild/stretch-11#50 with Debian stretch on armel"
JAVA_PATCHES=" main nosflt lib new bkpt jfr cds"
```

* Buster:
```
# ev3dev-lang-java openjdk build metadata
JAVA_ORIGIN="git"
JAVA_REPO="https://github.com/openjdk/jdk11u.git"
JAVA_BRANCH="jdk-11.0.8+10"
JAVA_COMMIT="871fe6e553f9b35f348240eaa5faab39b5de4c1b"
JAVA_VERSION="11.0.8+10-ev3"
CONFIG_VM="client"
CONFIG_DEBUG="release"
CONFIG_VERSION="11"
CONFIG_PLATFORM="ev3"
CONFIG_MODULES="java.se,jdk.jdwp.agent,jdk.unsupported,jdk.management.agent,jdk.jartool,jdk.crypto.ec"
BUILDER_COMMIT="9856847ab129cb9eb448b8e96562583778d76d86"
BUILDER_EXTRA="Jenkins eljbuild/buster-11#48 with Debian buster on armel"
JAVA_PATCHES=" main nosflt lib new bkpt jfr cds"
```

It was not possible to use the upstream packaging because the dependency on `ca-certificates-java` is still broken there.

> Tell me, what is the steps to rollback to previous package version?
> **I would like to do it**

If a new package needs to be built: The process is documented in https://github.com/ev3dev-lang-java/openjdk-ev3/tree/master/docs. You will need to build both the OpenJDK runtime and the package (both for stretch and buster) for the reasons above. For the building itself, you will need to use https://github.com/ev3dev-lang-java/openjdk-ev3/commit/9abe1a1de50021db286aaec34f7a737453a859c0 with which the previous package was likely built. You will also need to force the openjdk repo tag to ` jdk-11.0.6+10`, this needs to be done in [config.sh](https://github.com/ev3dev-lang-java/openjdk-ev3/blob/master/scripts/config.sh) by setting `VERSION_POLICY` to the corresponding tag. The resulting debian packages will however then either need to be versioned as `1:11.0.6~10(-1/2)` (incremented epoch number) or `11.0.8~10(-3/4)` (incremented revision numbers with the major version kept), so that currently running bricks upgrade to the new package.

> For the future:
> 
>     * How to check the checksum from the package generated by the pipeline in compare to the package stored into the `ev3dev Debian repository`? I would like to make a test to check that the exact binary is published on the repository.

After the packages downloaded directly from https://ci.adoptopenjdk.net/view/ev3dev/job/eljbuild/job/stretch-11/ are used, it should be enough to compare the Jenkins `.deb` MD5 hash to the hash contained in http://archive.ev3dev.org/debian/dists/stretch/main/binary-armel/Packages, against which the `.deb` files are validated.

> **Tell me if it is clear the actions to do, you are abusing about the generosity to collaborate in the Java project.**

Yes, I am abusing your generosity, but at the same it a) fixes a issue it causes in the ev3dev image generation process, and b) fixes a bug.

**Comment by jabrena** [comment]
Hi @JakubVanek,

This the issue that you said that it is solved was not merged in Master:
https://github.com/ev3dev-lang-java/openjdk-ev3/pulls

So, I repeat again, it is necessary to rollback to previous version.

If any user require `https` connections exist a workaround, use a complete JDK, it is simple.
You are not authorised to publish any Debian package related to Java, so yes, you are abusing and don´t continue that you behave as a Hero, because you don´t follow any rule, you go in your way.

Do you have any backup about the previous package?

Juan Antonio

**Comment by JakubVanek** [comment]
> If any user require `https` connections exist a workaround, use a complete JDK, it is simple.

I do not care about the ev3dev-lang-java bug itself that much (as you say, it has an OK solution), I do care about the breakage of the ev3dev image generation process more.

> Do you have any backup about the previous package?

No, I do not have a backup of it.

**Comment by JakubVanek** [comment]
> No, I do not have a backup of it.

Moreover, such a backup wouldn't be trustworthy, as at that time the debian package was not built on AdoptOpenJDK Jenkins yet.

**Comment by jabrena** [comment]
Oki, we will use latest `Debian package` built by the pipeline:
https://ci.adoptopenjdk.net/view/ev3dev/job/eljbuild/job/stretch-11/lastSuccessfulBuild/artifact/build/pkg/jri-11-ev3_11.0.8~10-1_armel.deb

This is simple.

What is the next steps?

**Comment by JakubVanek** [comment]
Please see https://github.com/ev3dev-lang-java/openjdk-ev3/blob/master/docs/Packaging_signing_uploading.md

**Comment by JakubVanek** [comment]
You may also need to change the maintainer email in the following places:
https://github.com/ev3dev-lang-java/openjdk-ev3/blob/master/packaging/debian/control
https://github.com/ev3dev-lang-java/openjdk-ev3/blob/master/packaging/debian/changelog

I think it is required for the changelog, as `debsign` tries to find the key according to the email of the person who made the last changelog entry.

**Comment by jabrena** [comment]
Hi @dlech,

@JakubVanek abused with the process to publish Debian packages related to Java and I need to upload the latest packages genereated by the pipeline following the steps described in the documentation. So, I will need a SSH keypair.

Please, revoke the permissions from @JakubVanek, his behaviour is not trustable. 

Another alternative is if you can upload the Debian package directly:
https://ci.adoptopenjdk.net/view/ev3dev/job/eljbuild/job/stretch-11/lastSuccessfulBuild/artifact/build/pkg/jri-11-ev3_11.0.8~10-1_armel.deb

The action is not urgent, the official image is from:
https://github.com/ev3dev/ev3dev/releases/download/ev3dev-stretch-2020-04-10/ev3dev-stretch-ev3-generic-2020-04-10.zip

**Comment by jabrena** [comment]
in relation to the another package `libopenmpt`, where is the source code @JakubVanek?
I have few questions:

- Where is the source code and scripts related to generate the Debian package?

It will be necessary to store in a proper way.

Juan Antonio

**Comment by JakubVanek** [comment]
> in relation to the another package `libopenmpt`, where is the source code @JakubVanek?

https://github.com/ev3dev/libopenmpt, as linked in my first post.

> I have few questions:
> 
> * Where is the source code and scripts related to generate the Debian package?

https://github.com/ev3dev/libopenmpt/tree/ev3dev-stretch/debian, processed using https://www.ev3dev.org/docs/devtools/packaging-for-ev3dev/.

**Comment by jabrena** [comment]
> https://github.com/ev3dev/libopenmpt, as linked in my first post.

- What is the changes that you applied to that source code?
- Maybe you did some patch.

**Comment by JakubVanek** [comment]
> > https://github.com/ev3dev/libopenmpt, as linked in my first post.
> 
>  * What is the changes that you applied to that source code?
>  * Maybe you did some patch.

Please see the commit history. https://github.com/ev3dev/libopenmpt/commit/3dc5d930bfa6d1876fdf86336870184ee41dbe66

**Comment by JakubVanek** [comment]
But please note that if the jri-11-ev3 package being uploaded doesn't have a form of https://github.com/ev3dev-lang-java/openjdk-ev3/pull/64 or https://github.com/ev3dev-lang-java/openjdk-ev3/pull/61 (the second one was folded into https://github.com/ev3dev-lang-java/openjdk-ev3/pull/66), the half-configuration of packages will return unless there is somewhere another `apt-get install` later in the image generation process that would finish the package configuration.

**Comment by jabrena** [comment]
Ok, can you submit a `PR` to fix that issue? (Use master to make the changes, I don´t want to introduce any extra change in the PR. Describe how to test in the Brick and commands required. I will have to test the new Debian packaged generated)

At the end, this is the point, you published something not from the pipeline (the source of truth) and it was not verified, tested, traced and documented.

**Comment by JakubVanek** [comment]
> Ok, can you submit a `PR` to fix that issue?
> At the end, this is the point, you published something not from the pipeline (the source of truth) and it was not verified and documented.

* https://github.com/ev3dev-lang-java/openjdk-ev3/pull/66 is the solution discussed in https://github.com/ev3dev-lang-java/ev3dev-lang-java/issues/731. This means https://github.com/ev3dev-lang-java/openjdk-ev3/pull/61 + https://github.com/ev3dev-lang-java/openjdk-ev3/pull/62 + "minitest" job build phase that tests for the broker issue.

The sub-options of the solution (both approved and rejected) are the following reopened PRs:

Packaging (only one of these has to be applied + generating the `.deb` from `.tar.gz` with the fix is enough to fix this):
* https://github.com/ev3dev-lang-java/openjdk-ev3/pull/61 - this fixes the half-configuration by dropping the dependency on `ca-certificates-java` and using the CA certs from the tarball directly instead. This will keep them fixed at this version and they will not be updated automatically (unless with a new JRI package).
* https://github.com/ev3dev-lang-java/openjdk-ev3/pull/64 - this fixes the half-configuration by instead fixing the dependency on `ca-certificates-java`. This keeps the CA certificates on the brick recent by relying on the CA certificate update mechanism present in Debian.

JDK build (only one of these can be applied, but it does not have to be, image generation will work fine without it + the JDK runtime package needs to be rebuilt, as this changes the `.tar.gz` contents):
* https://github.com/ev3dev-lang-java/openjdk-ev3/pull/62 - this replaces the currently obsolete AdoptOpenJDK cacerts (not used by their builds, but present in their repo) by using Mozilla CA certificates
* https://github.com/ev3dev-lang-java/openjdk-ev3/pull/65 - this replaces the AdoptOpenJDK cacerts by instead keeping the stock OpenJDK CA certificates and not replacing them with AdoptOpenJDK ones.

minitest does not have its own PR as this solution was rejected.

**Comment by JakubVanek** [comment]
> Describe how to test in the Brick and commands required. I will have to test the new Debian packaged generated)
For the half-configured package:
```
sudo apt autoremove --purge jri-11-ev3 ca-certificates-java
sudo apt install ./jri-11-ev3* # new debian package
```
If the installation process does not fail with the error like in https://github.com/ev3dev-lang-java/openjdk-ev3/pull/61#issuecomment-633708427, then the packaging problem was fixed successfully.

For the SSL issue, running broker as described in the original issue or running [minitest](https://github.com/JakubVanek/elj-jdk-private-dev/tree/bugfix/ssl-support/minitest) should be enough to confirm the fix.

**Comment by jabrena** [comment]
Sorry @JakubVanek,

that PR are not required. I only need the PR to use that .deb to publish.

> But please note that if the jri-11-ev3 package being uploaded doesn't have a form of ev3dev-lang-java/openjdk-ev3#64 or ev3dev-lang-java/openjdk-ev3#61 (the second one was folded into ev3dev-lang-java/openjdk-ev3#66), the half-configuration of packages will return unless there is somewhere another apt-get install later in the image generation process that would finish the package configuration.

I closed previous PR and in relation to:
https://github.com/ev3dev-lang-java/openjdk-ev3/pull/66

I am waiting for your OpenJDK test suite.

So please, publish a PR with the minimum changes for the Debian format in order to publish in the repository.

I am not sure If I have to write in Cech language to be more explicit.

Juan Antonio

**Comment by JakubVanek** [comment]
> So please, publish a PR with the minimum changes for the Debian format in order to publish in the repository.

This is why I included the suboptions:

> The sub-options of the solution (both approved and rejected) are the following reopened PRs:
> 
> Packaging (only one of these has to be applied + generating the `.deb` from `.tar.gz` with the fix is enough to fix this):
> 
>  * [ev3dev-lang-java/openjdk-ev3#61](https://github.com/ev3dev-lang-java/openjdk-ev3/pull/61) - this fixes the half-configuration by dropping the dependency on `ca-certificates-java` and using the CA certs from the tarball directly instead. This will keep them fixed at this version and they will not be updated automatically (unless with a new JRI package).
> 
>  * [ev3dev-lang-java/openjdk-ev3#64](https://github.com/ev3dev-lang-java/openjdk-ev3/pull/64) - this fixes the half-configuration by instead fixing the dependency on `ca-certificates-java`. This keeps the CA certificates on the brick recent by relying on the CA certificate update mechanism present in Debian.

**Comment by jabrena** [comment]
I will review it

**Comment by dlech** [comment]
> So, I will need a SSH keypair.

You will need to generate your own SSH key for uploading and GPG key for signing and send both of the public keys to me.

**Comment by dlech** [comment]
I have a backup of `jri-11-ev3_11.0.6~10-2_armel.deb` if we need to restore it. However it sounds like you have worked out something else.

**Comment by dlech** [comment]
Going forward (ev3dev-buster and beyond), it would be really nice if this package could be packaged like all of the other ev3dev packages that include a Debian source package (`.dsc`) so that the source code can be easily obtained for licensing purposes and so that the package can easily be rebuilt by anyone.

**Comment by JakubVanek** [comment]
> I have a backup of `jri-11-ev3_11.0.6~10-2_armel.deb` if we need to restore it. However it sounds like you have worked out something else.

I have found that it might be useful after all. Do you have also the `jri-11-ev3_11.0.6~10-1_armel.deb` from Stretch? Then there are the `jri-11-ev3-dbgsym` packages, but they are not that useful.

It seems I'm a moron and I have not checked what happens during the ev3dev build process. https://travis-ci.org/github/ev3dev/docker-library/jobs/673535028#L2962 For some reason, the `jri-11-ev3` package gets correctly configured before `ca-certificates-java` and the issue in https://github.com/ev3dev-lang-java/openjdk-ev3/pull/61#issuecomment-633708427 that I wanted to fix with the unauthorized upload does not manifest, as this way `ca-certficates-java` gets its `java` binary before it needs to use it. There is something with the cacerts missing, that might or might not be related. I remembered I saw `apt-get install` of an unrelated package on ev3dev finish the configuration of `jri-11-ev3`, but it seems this was either not the case or it was not a clean image.

As the old 11.0.6 package does not appear to break the image build process, I have no problem with it being restored. Then it also seems that current unpatched https://github.com/ev3dev-lang-java/openjdk-ev3 should produce a valid package too.

I would like to apologize for causing this chaos. I should have checked this before assuming the circumstances of the bug. My upload of the package was also a bad idea, but it was a path of least resistance given that I do not want to contribute to ev3dev-lang-java more.

The revocation has to go through in any case.

**Comment by dlech** [comment]
> Do you have also the `jri-11-ev3_11.0.6~10-1_armel.deb` from Stretch?

There are weekly backups going back until August 1. This includes the debug symbols packages as well.

**Comment by JakubVanek** [comment]
> Going forward (ev3dev-buster and beyond), it would be really nice if this package could be packaged like all of the other ev3dev packages that include a Debian source package (`.dsc`) so that the source code can be easily obtained for licensing purposes and so that the package can easily be rebuilt by anyone.

In hindsight, this should be implementable now. With the current 32bit-compatible ARMv8 build servers in the AdoptOpenJDK farm, the upstream openjdk package likely could be adapted in such a way that a native build for EV3 is performed.

**Comment by JakubVanek** [comment]
> > Do you have also the `jri-11-ev3_11.0.6~10-1_armel.deb` from Stretch?
>
> There are weekly backups going back until August 1. This includes the debug symbols packages as well.

Thank you, this does give @jabrena more options of how to continue with the package.

**Comment by dlech** [comment]
In ev3dev-stretch, I have removed `jri-11-ev3_11.0.8~10-1_armel.deb` and `jri-11-ev3-dbgsym_11.0.8~10-1_armel.deb` and restored `jri-11-ev3_11.0.6~10-1_armel.deb:` and `jri-11-ev3-dbgsym_11.0.6~10-1_armel.deb`.

In ev3dev-buster, I have removed `jri-11-ev3_11.0.8~10-2_armel.deb` and `jri-11-ev3-dbgsym_11.0.8~10-2_armel.deb` and restored `jri-11-ev3_11.0.6~10-2_armel.deb:` and `jri-11-ev3-dbgsym_11.0.6~10-2_armel.deb`.

**Comment by jabrena** [comment]
Many thanks @dlech

**Comment by JakubVanek** [comment]
@dlech could you also please revoke my access to the SFTP server? I don't know if I still have write access to the ev3dev-lang-java packages, but I can still run
```
$ sftp ev3dev-upload@reprepro.ev3dev.org
Connected to reprepro.ev3dev.org.
sftp> 
```

If I create some own packages (own as in not part of a project of someone else) for inclusion ev3dev in the future, I will create a new issue asking for the access or for the individual uploads.

EDIT: regarding own packages: I do not plan to manage a package on behalf of other projects.

**Comment by dlech** [comment]
I have removed your SSH key from the server.

**Comment by JakubVanek** [comment]
I can confirm this, the server refuses my connection now.
```
$ sftp ev3dev-upload@reprepro.ev3dev.org
ev3dev-upload@reprepro.ev3dev.org: Permission denied (publickey).
Connection closed
```

**Comment by jabrena** [comment]
Thanks, now the manual process executed by @JakubVanek is stopped and it is clear what Debian packages were published for current build: https://github.com/ev3dev/ev3dev/releases/download/ev3dev-stretch-2020-04-10/ev3dev-stretch-ev3-generic-2020-04-10.zip and if they are valid for next builds.

@JakubVanek, I would like to review what is the steps required to generate the build for the next build, for Java 15. 
I would like to do by myself using the documentation provided by you. What is the steps?

- [ ] Patches
- [ ] Scripts
- [ ] Pipelines
- [ ] Documentation
- [ ] Testing
- [ ] Deployment

Juan Antonio

**Comment by JakubVanek** [comment]
Parts of this are under https://github.com/ev3dev-lang-java/openjdk-ev3/blob/master/docs/Contents.md.
 * Rebased patches for jdk15 are already there. For newer releases, they need to be copied and rebased, some commands are in [docs](https://github.com/ev3dev-lang-java/openjdk-ev3/blob/master/docs/Included_patches.md#patch-rebasing); the patches need to be applied in the order they are listed in [fetch.sh](https://github.com/ev3dev-lang-java/openjdk-ev3/blob/9856847ab129cb9eb448b8e96562583778d76d86/scripts/fetch.sh#L101).
 * scripts - a new JDK15 section needs to be added in [config.sh](https://github.com/ev3dev-lang-java/openjdk-ev3/blob/master/scripts/config.sh), based on the JDK14 and JDK-Master sections.
 * pipelines - you need to copy the existing job and change its default parameters, [docs](https://github.com/ev3dev-lang-java/openjdk-ev3/blob/master/docs/AdoptOpenJDK_Jenkins.md) have information regarding the parameters, but https://github.com/ev3dev-lang-java/openjdk-ev3/blob/master/Jenkinsfile is the ultimate source of truth
 * documentation - [README.md](https://github.com/ev3dev-lang-java/openjdk-ev3/blob/master/README.md) needs to be updated
 * testing - bootcycle build is performed on the build server to check basic functionality; for EV3, running some ev3dev-lang-java project on the brick should do
 * deployment - documented [here](https://github.com/ev3dev-lang-java/openjdk-ev3/blob/master/docs/Packaging_signing_uploading.md); however, as the `.deb` will be taken from Jenkins, docker will not be used

**Comment by JakubVanek** [comment]
However, I think going through the sources of and maybe rewriting http://github.com/ev3dev-lang-java/openjdk-ev3 and https://ci.adoptopenjdk.net/view/ev3dev/ would be a good idea. That way, you will become familiar with the internals of the build process and further development will become possible. I wouldn't immediately switch to the Debian-native packaging as that requires reviewing the build process as a whole, but it could be done in a kind-of "second pass".

**Comment by jabrena** [comment]
Sorry @JakubVanek, if every developer rewrite any component, the project would be maintained in a bad way. 
I will maintain current process only, the CI status is Green. I know that you love rewrite stuff, but it is not the case. I will maintain current process and I will update the documentation to be easier not only for me, for other guys in the future. Thanks for the comment.
